mediacloud/mediawords/dbi/downloads.py

```diff
@@ -0,0 +1,388 @@
+"""Various helper functions for downloads, including storing and fetching content.
+
+This module includes various helper function for dealing with downloads.
+
+Most importantly, this module has the store_content and fetch_content
+functions, which store and fetch content for a download from the pluggable
+content store.
+
+The storage module is configured in mediawords.yml by the
+mediawords.download_storage_locations setting.
+
+The three choices are:
+
+* 'postgresql', which stores the content in a separate postgres table and
+  optionally database
+* 'amazon_s3', which stores the content in amazon_s3
+* 'databaseinline', which stores the content in the downloads table downloads
+  are no longer stored in `databaseinline', only read from.
+
+The default is 'postgresql', and the production system uses Amazon S3.
+
+This module also includes extract and related functions to handle download
+extraction.
+"""
+
+import random
+import re
+import typing
+
+from mediawords.db import DatabaseHandler
+from mediawords.key_value_store import KeyValueStore
+import mediawords.util.extract_text
+import mediawords.util.html
+from mediawords.util.log import create_logger
+
+log = create_logger(__name__)
+
+# PostgreSQL table name for storing raw downloads
+RAW_DOWNLOADS_POSTGRESQL_KVS_TABLE_NAME = 'raw_downloads'
+
+# Min. content length to extract (assuming that it has some HTML in it)
+MIN_CONTENT_LENGTH_TO_EXTRACT = 4096
+
+# these are initialized by calling the various get_*_story() functions below
+_inline_store = None
```

Comment:
I poked around in the world of pythonicism for a bit and couldn't find this addressed anywhere.  What is the argument for putting globals instead classes instead of just within packages?  Why mediawords.dbi.downloads.Globals.MIN_CONTENT_LENGTH_TO_EXTRACT rather than mediawords.dbi.downloads.MIN_CONTENT_LENGTH_TO_EXTRACT?

Reply:
We could (should) definitely self-contain global variables, but IMHO Hal did the right thing by not doing it now. As per our Perl -> Python rewrite strategy, we rewrite first and refactor later because those are two different tasks, and the latter task is much more easily done while in Python due to various fancy tooling being available to us (for comparison, Perl doesn't even have a proper IDE).